intellij-plugin-verifier/verifier-repository/src/main/java/com/jetbrains/pluginverifier/repository/repositories/marketplace/MarketplaceRepository.kt

```diff
@@ -1,226 +1,123 @@
+    updateMetadata.version,
+    updateMetadata.since.prepareIdeVersion(),
+    updateMetadata.until.prepareIdeVersion(),
+    updateMetadata.vendor,
+    parseSourceCodeUrl(updateMetadata.sourceCodeUrl),
+    getBrowserUrl(pluginId),
+    updateId,
+    getDownloadUrl(updateId),
+    updateMetadata.tags
+  )
 
   /**
    * Given the [update ID] [updateId], which is a unique identifier
    * of the plugin version in the Plugins Repository database,
    * returns its more detailed [UpdateInfo].
    */
-  fun getPluginInfoById(updateId: Int) =
-    updateInfosRequester.getUpdateInfoById(updateId, DEFAULT_BATCH_REQUEST_SIZE)
-
-  override fun getLastCompatibleVersionOfPlugin(ideVersion: IdeVersion, pluginId: String) =
-    getAllVersionsOfPlugin(pluginId).filter { it.isCompatibleWith(ideVersion) }.maxBy { it.updateId }
-
-  override fun getAllVersionsOfPlugin(pluginId: String): List<UpdateInfo> =
-    try {
-      repositoryConnector
-        .getPluginUpdates(pluginId).executeSuccessfully().body()!!
-        .updateIds
-        .mapNotNull { updateInfosRequester.getUpdateInfoById(it.updateId, DEFAULT_BATCH_REQUEST_SIZE) }
-    } catch (e: Exception) {
-      e.rethrowIfInterrupted()
-      emptyList()
-    }
-
-  override fun getLastCompatiblePlugins(ideVersion: IdeVersion) =
-    repositoryConnector.getAllCompatibleUpdates(ideVersion.asString())
-      .executeSuccessfully().body()!!
-      .map { updateInfosRequester.putJsonUpdateInfo(it) }
-
-  override fun getIdOfPluginDeclaringModule(moduleId: String) =
-    INTELLIJ_MODULE_TO_CONTAINING_PLUGIN[moduleId]
-
-  override fun toString() = "Plugin Repository " + repositoryURL.toExternalForm()
-
-  /**
-   * This class is responsible for requesting the [UpdateInfo]s.
-   */
-  private inner class UpdateInfosRequester {
-
-    /**
-     * Because of possible changes of *since, until*
-     * values in the repository's database,
-     * it is necessary to occasionally update the caches.
-     * Thus, the values are invalidated every hour.
-     *
-     * The value of the cache is `Optional` because
-     * the guava's caches don't allow to cache `null` values.
-     */
-    private val updateInfos = CacheBuilder
-      .newBuilder()
-      .expireAfterWrite(1, TimeUnit.HOURS)
-      .build<Int, Optional<UpdateInfo>>()
+  fun getPluginInfoById(updateId: Int): UpdateInfo? {
+    val pluginUpdateBean = pluginRepositoryInstance.pluginUpdateManager.getUpdateById(updateId) ?: return null
+    return requestAndCreateUpdateInfo(pluginUpdateBean.pluginId, updateId)
+  }
 
-    //synchronized block is used here to avoid
-    //performing unnecessary interleaving batch requests
-    @Synchronized
-    fun getUpdateInfoById(updateId: Int, batchSize: Int): UpdateInfo? =
-      updateInfos
-        .get(updateId) { requestUpdateInfo(updateId, batchSize) }
-        .orElse(null)
+  private fun getBrowserUrl(pluginId: Int) =
+    URL("${repositoryURL.toExternalForm().trimEnd('/')}/plugin/$pluginId")
```

Comment:
Please don't use obsolete endpoint `/plugin/{pluginId}`. use new one from the rest-client library